pkg/receive/writable_store.go

```diff
@@ -0,0 +1,40 @@
+package receive
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	"github.com/prometheus/prometheus/prompb"
+	"github.com/prometheus/prometheus/storage"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/thanos-io/thanos/pkg/store"
+	"github.com/thanos-io/thanos/pkg/store/storepb"
+)
+
+type ReadWriteStore struct {
+	*store.TSDBStore
+	w *Writer
+}
+
+// NewReadWriteStore creates a tsdb store that's also writable.
+func NewReadWriteStore(r *store.TSDBStore, w *Writer) *ReadWriteStore {
+	return &ReadWriteStore{
+		TSDBStore: r,
+		w:         w,
+	}
+}
+
+func (rw *ReadWriteStore) RemoteWrite(ctx context.Context, r *storepb.WriteRequest) (*storepb.WriteResponse, error) {
```

Comment:
Right now if anything is received on the gRPC service it will _always_ be written to the local disk without checking if it should be forwarded. Once we choose to have clients directly remote write gRPC to a receive endpoint, this will no longer work for us.

Reply:
Yeah I need to do some more changes to account for that. I’ll continue that part tomorrow